src/components/PokemonList.tsx

```diff
@@ -1,51 +1,85 @@
-import React, { useState } from 'react';
+import React from 'react';
 import PokemonCard from './PokemonCard';
-import { usePokemonData } from '../hooks/usePokemonData';
+import { usePokemons } from '../hooks/queries/usePokemons';
 import Loading from './Loading';
 import ErrorMessage from './ErrorMessage';
 import { ERROR_MESSAGES } from '../helpers/constants';
 import './PokemonList.css';
 import useFilter from '../hooks/useFilter';
+import { BeatLoader } from 'react-spinners';
+import { Pokemon } from '../types/Pokemon';
 
 const PokemonList: React.FC = () => {
-  const { pokemonList, loading, error, loadMorePokemon } = usePokemonData();
   const { filter, clearFilter } = useFilter();
-  const [buttonClicked, setButtonClicked] = useState(false);
+
+  // Using infinite query.
+  const {
+    data,
+    isLoading,
+    isError,
+    hasNextPage,
+    fetchNextPage,
+    isFetchingNextPage,
+  } = usePokemons();
```

Comment:
Talvez nem tanto pra esse app, mas é comum renomear essas variáveis e métodos, principalmente se tiver usando mais de uma query diferente pra diferenciar:

```suggestion
  const {
    data: pokemons,
    isLoading: isLoadingPokemons,
    // etc...
    isError,
    hasNextPage,
    fetchNextPage,
    isFetchingNextPage,
  } = usePokemons();
```

Acho que só renomear o `data` pra `pokemons` ou `pokemonsData` ou outra coisa já ficaria mais claro, mas em casos em que vc depende de múltiplos `isLoading` diferentes por exemplo, é interessante renomear eles na hora de usar.